ops/model.py

```diff
@@ -492,7 +492,9 @@ def _get_unique(self, relation_name, relation_id=None):
                 is_peer = relation_name in self._peers
                 return Relation(relation_name, relation_id, is_peer,
                                 self._our_unit, self._backend, self._cache)
-        num_related = len(self[relation_name])
+        relations = self[relation_name]
+        num_related = len(relations)
+        self._backend._validate_relation_access(relation_name, relations)
```

Comment:
It still doesn't feel great to have this line in the production facing parts of this code.

But as we discussed in the sync, this is a lightweight call, to address a specific problem which is difficult to address by other means. I am +1 on this approach.

Reply:
As you say, this does feel a bit dirty, but I haven't been able to come up with a less bad solution.  If anyone can think up something better I'm all ears!